lib/bolt/plugin/task.rb

```diff
@@ -33,9 +33,9 @@ def inventory
         @inventory ||= Bolt::Inventory.new({}, config)
       end
 
-      def run_task(opts)
+      def run_task(opts, target = 'localhost')
         result = pal.run_task(opts['task'],
-                              'localhost',
+                              target,
```

Comment:
I'm concerned that a Target object bound to one Inventory instance is being used with a different Inventory instance here.